src/wp-includes/class-wp-block.php

```diff
@@ -88,7 +113,7 @@ class WP_Block {
 	 * @since 5.5.0
 	 * @var array
 	 */
-	public $inner_content = array();
+	public $inner_content;
```

Comment:
I am wondering if changing the default type could lead to back-compat problems. `null` is still falsey but if a plugin is doing strict type checking then it might be problematic.

Once set, can any of these values be an `empty()` value? 

Maintaining the `__get()` could solve this problem.

Reply:
There shouldn't be any compatibility problems here because all of the attributes are initialised in `WP_Block::__construct()` by default. The only change in behaviour is if `'is_eager' => false` is provided.